src/systems/renderer.js

```diff
@@ -21,7 +21,10 @@ module.exports.System = registerSystem('renderer', {
     colorManagement: {default: false},
     gammaOutput: {default: false},
     alpha: {default: true},
-    foveationLevel: {default: 0}
+    foveationLevel: {default: 0},
+    xrSessionInit: {default: {
```

Comment:
We don't have proper support for arbitrary objects in the schema. What about doing to `properties` instead:
```
webXRrequiredFeatures: {type:'array', default: ['local-floor']},
webXROptionalFeatures: {type:'array', default: ['bounded-floor'])}
```

Reply:
Also wondering if we should move this out of `renderer` component and move it to a `webxr` one. It might be cleaner.

Reply:
e.g: `<a-scene webxr="requiredFeatures: local-floor, dom-overlay, optionalFeatures: bounded-floor, dom-overlay>`

Reply:
That works for those two specific arrays, but we still need an attachment point where systems can modify the raw init argument, for example setting `domOverlay.root`.

I've updated this, what do you think?

Reply:
About your example, I think it's a bit clunky and not very future-proof if scenes need to maintain an explicit list of features. Having a system such as _dom-overlay_ automatically add to the arrays may be a bit of an abstraction layer break, but I think it's a lot more convenient, and makes it easier for people to mix&match systems and components in the future.

I agree that this would be cleaner in a _webxr_ component, but maybe use names such as "requiredReferenceSpaces" and "optionalReferenceSpaces" to make it clearer that this will be used as input to creating the XR init dictionary, without implying that it'll be an exhaustive list?

Reply:
we could have ` webxr` systems that has an internal reference to the session object that others could access for experimentation. It seems that `renderer` and `webxr` are different things and worth decoupling. Maybe something like this:

```
module.exports.System = registerSystem('webxr', {
  schema: {
     requiredFeatures: {type: 'array', default: ['local-floor']},
     optionalFeatures: {type: 'array', default: ['bounded-floor']},
     overlayElement: {type: 'selector'}
  },
 update: function () {
    var data =  this.data;
    this.sessionConfiguration = {
        requiredFeatures: data.requiredFeatures,
        optionalFeatures: data.optionalFeatures
     }

     if (data.overlayElement) {
        this.sessionConfiguration.domOverlay = {root: data.overlayElement};
     }
  }
});
```
Experimentation can happen by accessing `sceneEl.systems.webxr.sessionConfiguration`

Reply:
Done, using your suggested webxr system.